### PR TITLE
Gray-out zeros in Daily Form Activity and Submissions By Form reports

### PR DESCRIPTION
**Pull request by kaapstorm**
(FB 147530) I'd like to highlight outliers based on median absolute deviation, as Cory suggested. The relevant data is down columns instead of across rows, so if the styling is done in the HTML (as opposed to calculated by the browser) the table would need to be created twice; once for the data and a second pass for the styling. That's a bit more code than this little change, so I reckon we start with this, and add that as a nice to have for later (or sooner).

Reports with this new styling look like so:
gray-zero

I've used standard "commCareGray". We might want to lighten it a little to make the difference more pronounced.

WARNING: (Where's that HTML "blink" tag when you need it? :-) ) After arguing with my local environment for too long trying to get a Daily Form Activity report to show some data, I threw in the towel (admittedly feeling too sheepish to ask for more help doing something that I'm sure is supposed to be obvious). (`./manage.py run_ptop --all` didn't seem to resolve the problem.) I'm happy to go back to it later, but I wanted to get a PR out. So my warning is that I have NOT TESTED this code change with the Daily Form Activity report. I'm pretty sure my one-line change in get_row() isn't going to break stuff, but if we can watch it working on Staging, I'd be more comfortable.

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by czue** [comment]
FYI, I tested this locally with some data and I don't see any style difference. The class is getting added to rows but it is not having any impact on the data itself.

image

image

(Also what's up with the empty `class` attributes on all the nonzero tags? might be a legacy issue)

**Comment by czue** [comment]
(saw this behavior on daily form activitiy and submissions by form)

**Comment by kaapstorm** [comment]
It might be that the browser is using a cached stylesheet, so ignoring the "gray-zero" class. The empty class tags are interesting -- do you only get them on the Daily Form Activity report? I don't see them on my Submissions By Form report, but I do see "<title>" tags after all the values.

**Comment by kaapstorm** [comment]
My mistake. I do have the empty `class` attributes, but it seems the browser is adding them. (Both Firefox and Chrome.) I checked the response from the server, and table cells look like 
```html
    <td>    <span class="gray-zero">0</span>    <span title="0"></span></td>
    <td>    124    <span title="124"></span></td>
    <td>    97    <span title="97"></span></td>
    <td>    <span class="gray-zero">0</span>    <span title="0"></span></td>
```

Inspecting it on Chrome shows it as something like:
td_class_span_title 

Maybe JS is adding the td class.

File: corehq/apps/reports/generic.py [hunk 1]
```diff
@@ -943,10 +943,11 @@ def report_context(self):
             context.update(provider_function(self))
         return context
 
-    def table_cell(self, value, html=None):
+    def table_cell(self, value, html=None, zerostyle=False):
+        styled_value = '<span class="gray-zero">0</span>' if zerostyle and value == 0 else value
```
Comment (biyeun) [inline corehq/apps/reports/generic.py]: not a fan of using html in python code, but I realize that the reports are currently structured in a way that you can't _really_ work around that. Sigh. x_x

(non-actionable...just a comment on the current structure which originally was my fault :crying_cat_face:)

Comment (biyeun) [inline corehq/apps/reports/generic.py]: have you looked at the `muted` class from twitter bootstrap 2.3.2

Comment (kaapstorm) [inline corehq/apps/reports/generic.py]: +1 about HTML in Python. :-/ Also less than happy with using the same HTML twice in this PR, but couldn't think of a clean way around that. I really like your `muted` suggestion. I'll try that, and drop the change to tables.less.

Comment (czue) [inline corehq/apps/reports/generic.py]: I never figured out why I didn't see the changes locally, but happy to try as soon as you've updated it.

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by kaapstorm** [comment]
Gray zeros using "mute" instead of defining a new style:
grey_zeros_using_mute

**Comment by czue** [comment]
looks good!

**Comment by sravfeyn** [comment]
This made columns unsortable in the table. Will fix.

**Comment by kaapstorm** [comment]
Sravan, if it's any help, you'll see that in generic.py table cells use a sort key (so the Submissions By Form report should be sortable), but they don't in monitoring.py. Maybe using the same approach in monitoring.py will fix the problem in the Daily Form Activity report.

**Comment by sravfeyn** [comment]
Hey, sorry, it does work. It was `Loading` forever when I tried sort, but if I select to show all rows and click sort, it sorts. Not reproducible.

**Pull request merged**